File: adventOfCode22/typescript/day1.ts
import { readFileSync } from 'fs';

const file = readFileSync('../resources/day1.txt','utf8')

let elfCalories = <number[]>[]
let currentElf = 0
file.split(/\r?\n/).forEach(function(line){
    if(line === '') {
        elfCalories.push(currentElf)
        currentElf = 0
    } else {
        currentElf += Number.parseInt(line)
    }
  })

elfCalories = elfCalories.sort((n1,n2) => (n1 - n2) * -1);
console.log(`Solution A: ${elfCalories[0]}`)
console.log(`Solution B: ${elfCalories[0] + elfCalories[1] + elfCalories[2]}`)

/**
 * Solutions
 * 
 * a: 69883
 * b: 207576
 */

File: adventOfCode22/golang/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	readFile, err := os.Open("../resources/day1.txt")

	check(err)

	fileScanner := bufio.NewScanner(readFile)

	fileScanner.Split(bufio.ScanLines)

	var elfCalories []int
	currentElf := 0

	for fileScanner.Scan() {
		line := fileScanner.Text()

		if line == "" {
			elfCalories = append(elfCalories, currentElf)
			currentElf = 0
		} else {
			calories, err := strconv.Atoi(line)
			check(err)

			currentElf += calories
		}
	}

	sort.Sort(sort.Reverse(sort.IntSlice(elfCalories)))
	fmt.Println("1st: ", elfCalories[0])
	fmt.Println("2st: ", elfCalories[0]+elfCalories[1]+elfCalories[2])

	readFile.Close()
}
